### PR TITLE
Add datadog component documentation

### PR DESCRIPTION
**Pull request by nunofgs**
**Description:**
Adds documentation, screenshots and logo for the proposed `datadog` component.

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant):** home-assistant/home-assistant#7158

### CODE REVIEW

File: source/_components/datadog.markdown [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# Example configuration.yaml entry
+datadog:
+```
+
+Configuration variables:
+
+- **host** (*Optional*): IP address of your Datadog host, eg. 192.168.1.10. Defaults to `localhost`.
+- **port** (*Optional*): Port to use. Defaults to 8125.
+- **prefix** (*Optional*): Prefix to use. Defaults to `hass`.
+- **rate** (*Optional*): The sample rate. Defaults to 1.
```
Comment (Landrash) [inline source/_components/datadog.markdown]: What does the rate describe? Bytes, Liters, Minutes etc?

File: source/_components/datadog.markdown [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+To use the `datadog` component in your installation, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+datadog:
+```
+
+Configuration variables:
+
+- **host** (*Optional*): IP address of your Datadog host, eg. 192.168.1.10. Defaults to `localhost`.
```
Comment (Landrash) [inline source/_components/datadog.markdown]: Suggest you change this to 
>  The IP address or hostname of your Datadog host, e.g. 192.168.1.23.  Defaults to `localhost`.

File: source/_components/datadog.markdown (hunk 1 above)
Comment (nunofgs) [inline source/_components/datadog.markdown]: It is the sample rate of UDP packets sent to Datadog. More information [here](https://help.datadoghq.com/hc/en-us/articles/208398693--dog-statsd-sample-rate-parameter-explained).

Comment (Landrash) [inline source/_components/datadog.markdown]: Oh. That one would be tad hard to simplify. Keep it as is.

**Comment by Landrash** [comment]
Looks good and can be merged after parent pr is merged.

**Pull request merged**